src/screens/splash.rs

```diff
@@ -35,48 +40,84 @@ pub(super) fn plugin(app: &mut App) {
         )
             .run_if(in_state(Screen::Splash)),
     );
+
+    // exit the splash screen early if the player hits escape
+    app.add_systems(
+        Update,
+        exit_splash_screen
+            .run_if(input_just_pressed(KeyCode::Escape).and_then(in_state(Screen::Splash))),
+    );
 }
 
 const SPLASH_BACKGROUND_COLOR: Color = Color::srgb(0.157, 0.157, 0.157);
 const SPLASH_DURATION_SECS: f32 = 1.8;
 const SPLASH_FADE_DURATION_SECS: f32 = 0.6;
 
-fn spawn_splash(mut commands: Commands, asset_server: Res<AssetServer>) {
-    commands
-        .ui_root()
-        .insert((
-            Name::new("Splash screen"),
+#[derive(Component)]
+struct SplashScreenContainer;
+
+#[derive(Resource)]
+struct SplashScreenImageList(VecDeque<&'static str>);
+
+impl Default for SplashScreenImageList {
+    fn default() -> Self {
+        // Use paths here rather than an ImageHandle as the loading screen hasn't
+        // run yet. To show your own splash images, replace or add here
+        Self(VecDeque::from_iter(["images/splash.png"]))
+    }
+}
```

Comment:
Can `impl FromWorld` instead, to get access to `world.resource::<AssetServer>()`. Then loading the images can be done here.